Shopify/ruby-lsp
# Add GitHub action to create release

### Motivation

As part of our effort to simplify our releases, we should try to automate as many steps as possible. Creating GitHub releases through the UI can be fully automated with an action, so I propose we do it.

### Implementation

Removed CHANGELOG.md and its entry in the gemspec files.

The new action is fired whenever a new tag starting with the name `v` is pushed, which is done by Shipit for us. Then we use github-scripts to create a new release using the current owner and repository. The `${{ github.ref }}` is the tag that triggered the workflow (e.g.: `v0.1.2`), which we use as both the tag for the release and for the title. Finally, we use the auto-generated release notes to create the release body.

### Manual Tests

I really wish we could test this and make sure it works before committing to it. I have this idea, but I'd like to hear if anyone has a better one

1. Create a fake version tag starting with `v` and push it
2. Verify if this actually works
3. Delete the release
4. Delete the tag
